### PR TITLE
Feature/classes endpoints and middleware for GETS, added swagger

### PR DESCRIPTION
**Pull request by katherineyevsukov**
## Description

- Endpoints to return classes or a specific class were updated based on new classes table which was recently updated. 
- Swagger was implemented for these endpoints to document changes. 
- Some bugs were fixed on classes middleware and error handling middleware so proper error responses are returned. 
- Ran formatter over whole repo

Fixes # (issue)

## Loom Video

https://www.loom.com/share/b70f00d94ace45dbaf09c2d2911552b0

## Type of change

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have removed unnecessary comments/console logs from my code
- [X] I have made corresponding changes to the documentation if necessary (optional)
- [X] My changes generate no new warnings
- [X] I have checked my code and corrected any misspellings
- [X] No duplicate code left within changed files
- [X] Size of pull request kept to a minimum
- [X] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by mcmillanj — APPROVED** [review]
great job having the Endpoints return a specific class based on new classes table which was recently updated, also great job with the implementing of swagger

**Review by 19jlevitre — APPROVED** [review]
Love how you cleaned and formatted everything. Love the many commits. This is useful, efficient and solves a problem, so looks good to me.

**Review by natecheney-dev — APPROVED** [review]
Great job on discovering and fixing the problems, updating the readme, and removing unnecessary commits to clean up the code.

**Review by ElchinNasirov — APPROVED** [review]

File: api/classInstances/classInstanceMiddleware.js [hunk 1]
```diff
@@ -8,14 +9,13 @@ const { findByInstructorId } = require('../instructor/instructorModel');
 const { getAllCourseTypes } = require('../courseTypes/courseTypesModel');
 
 const checkClassInstanceExist = async (req, res, next) => {
-  const class_id = req.params.class_id;
+  const class_id = parseInt(req.params.class_id);
```
Comment (ryan-hamblin) [inline api/classInstances/classInstanceMiddleware.js:12]: Another technique you could explore here is `Number()` or `String()` or even `.toString()`

**Review by ryan-hamblin — COMMENTED** [review]
This is a very clean pull request. The work here being done is definitely WIP as there are comments littered throughout that explain more or less where the development of this and future feature is at.

**Review by mcmillanj — APPROVED** [review]
you dd some cleaning up, great

**Pull request merged**